### DISCUSSION TITLE
Is there any available bootloader upgrade method if bootloader update scenarios talked in docs not satisfied?

### DISCUSSION THREAD

**Discussion by andyliu-666** [rauc/rauc#1194] [Q&A]
uboot + Symmetric A/B rootfs

Storage:
raw nor flash + mtd + mtdblock ----- for uboot
raw nand flash + mtd + mtdblock + UBI volume(no parition) --- for rootfs(ubifs) 
so methods talked in docs:
1. Update Bootloader in eMMC Boot Partitions  --- not satisfied
2. Update Bootloader Partition in MBR               --- not satisfied  
3. Update Bootloader Partition in GPT                --- not satisfied

CPU: stm32mp151
so methods talked in docs:
Update Raw Bootloader Partition with Fallback    --- not satisfied

Current my design:
use slot hook(Post-Install Hook of rootfs slot), uboot image not include in manifest.raucm, it packed in bundle only for  hook script.
so:
during update, after rootfs installed successful, the post-install hook excuted, in the hook script, uboot image updated.
all uboot update steps handled in the hook script, it is not  atomic(although some checks added to make sure it installed correctly)。

according my scenario, is there a better way to upgrade bootloader?
thanks very much!

**Comment by ejoerns** [comment]
@andyliu-666 If there is an atomic way to update the bootloader very much depends on the SoC's ROM loader and how it reads the bootloader. I am not sure about the STM32MP1 capabilities with regard to NOR since I have used it with eMMC only.
If the ROM loader can read the bootloader from a fixed non-configurable address only, then there probably is no good way to implement atomic updates. Updating NOR (non-atomic) is however possible with the `nor` slot type.

Is there a reason for preferring NOR/NAND over probably simpler eMMC?
